TwoPerformant/Feed/Model/Cron.php

```diff
@@ -0,0 +1,91 @@
+<?php
+namespace TwoPerformant\Feed\Model;
+
+use Magento\Catalog\Api\ProductRepositoryInterface;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\Api\FilterBuilder;
+use Magento\Store\Model\StoreManagerInterface;
+
+
+class Cron
+{
+    protected $productRepository;
+    protected $searchCriteriaBuilder;
+    protected $filterBuilder;
+    protected $storeManager;
+
+    public function __construct(
+        ProductRepositoryInterface $productRepository,
+        SearchCriteriaBuilder $searchCriteriaBuilder,
+        FilterBuilder $filterBuilder,
+        StoreManagerInterface $storeManager
+    ) {
+        $this->productRepository = $productRepository;
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+        $this->filterBuilder = $filterBuilder;
+        $this->storeManager = $storeManager;
+    }
+
+    public function export()
+    {
+        $date = $this->getLastWeekDate();
+        $items = $this->getProducts($date);
+        $this->writeToFile($items);
+    }
+
+    protected function getLastWeekDate()
+    {
+        $now = new \DateTime();
+        $interval = new \DateInterval('P1W');
+        $lastWeek = $now->sub($interval);
+        return $lastWeek;
+    }
+
+    public function getProducts($date)
+    {
+        $filters = [];
+
+        $filters[] = $this->filterBuilder
+            ->setField('created_at')
+            ->setConditionType('gt')
+            ->setValue($date->format('Y-m-d H:i:s'))
+            ->create();
+
+        $this->searchCriteriaBuilder->addFilters($filters);
+
+        $searchCriteria = $this->searchCriteriaBuilder->create();
+        $searchResults = $this->productRepository->getList($searchCriteria);
+        return $searchResults->getItems();
+    }
+
+    protected function writeToFile($items)
+    {
+        if (count($items) > 0) {
```

Comment:
Do I understand correctly that
* if no products were added in the last week, the file will stay the same as last week (let's say, 1000 products)
* if 2 products were added in the last week, the file will contain only these 2 products?

The name of "feed" might be confusing, as in 2Performant a feed is the source that feeds us product information; but we do not expect incremental updates, but rather to get the updated information of all the products, every time we query it.

Reply:
No. 

If no products were added last week, the file will stay the same. 
If there are new products, I think the feed is overwritten. I added this on my setup just because it crashed otherwise. 😕 It should be removed before merging.